charmcraft/commands/store/__init__.py

```diff
@@ -636,6 +636,31 @@ def _get_lib_info(*, full_name=None, lib_path=None):
         full_name=full_name, path=lib_path, lib_name=lib_name, charm_name=charm_name)
 
 
+def _get_libs_from_tree(charm_name=None):
+    """Get library info from the directories tree (for a specific charm if specified).
+
+    It only pays attention to the proper disk structures, if there.
+    """
+    local_libs_data = []
+
+    if charm_name is None:
+        base_dir = pathlib.Path('lib') / 'charms'
+        charm_dirs = sorted(base_dir.iterdir()) if base_dir.exists() else []
+    else:
+        base_dir = pathlib.Path('lib') / 'charms' / charm_name
+        charm_dirs = [base_dir] if base_dir.exists else []
+
+    for charm_dir in charm_dirs:
+        for v_dir in sorted(charm_dir.iterdir()):
+            if v_dir.is_dir() and v_dir.name[0] == 'v' and v_dir.name[1:].isdigit():
```

Comment:
This assumes directory length is >1. I'd suggest a regex match, maybe with list comprehension. e.g.
version_dirs = [x for x in charm_dir.iterdir() if re.match("^v[0-9]+$", x.name)]

Reply:
should it allow '.' or similar in version?

Reply:
No "." or similar is allowed in that version, it must be just an integer.

If the directory name length is > 1, and it's an integer, it succeeds. Name length can not be zero (the filesystem prevents that). If length is 1, it will just not succeed:

```
>>> name = 'v'
>>> name[1:].isdigit()
False
```

No need for a regex.

Reply:
I stand corrected :+1: